commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddLdapConnection.java

```diff
@@ -0,0 +1,212 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.GroovyXmlTransform;
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.Subtree;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.offline.OfflineCommand;
+import org.wildfly.extras.creaper.core.offline.OfflineCommandContext;
+import org.wildfly.extras.creaper.core.online.CliException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+/**
+ * Add a new LDAP outbound connection.
+ */
+public final class AddLdapConnection implements OnlineCommand, OfflineCommand {
+
+    private final String connectionName;
+    private final List<String> handlesReferralsFor;
+    private final String initialContextFactory;
+    private final String referrals;
+    private final String searchCredential;
+    private final String searchDn;
+    private final String securityRealm;
+    private final String url;
+    private final boolean replaceExisting;
+    private final Map<String, String> properties;
+
+    private AddLdapConnection(Builder builder) {
+        this.connectionName = builder.connectionName;
+        this.handlesReferralsFor = builder.handlesReferralsFor;
+        this.initialContextFactory = builder.initialContextFactory;
+        this.referrals = builder.referrals;
+        this.searchCredential = builder.searchCredential;
+        this.searchDn = builder.searchDn;
+        this.securityRealm = builder.securityRealm;
+        this.url = builder.url;
+        this.replaceExisting = builder.replaceExisting;
+        this.properties = builder.properties;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws CliException, CommandFailedException, IOException,
+            TimeoutException, InterruptedException {
```

Comment:
Why do all the `apply` methods still have this crappy list of thrown exception? The `*Command` interface looked like this a long time ago, but now the interface has `throws Exception` and in any case, you should only declare those exceptions that the `apply` method can actually throw.